src/AltinnCore/Runtime/RestControllers/ProcessController.cs

```diff
@@ -41,29 +66,445 @@ public class ProcessController : ControllerBase
+            [FromRoute] string org,
+            [FromRoute] string app,
+            [FromRoute] int instanceOwnerId,
+            [FromRoute] Guid instanceGuid,
+            [FromQuery] string startEvent)
+        {
+            Instance instance = await instanceService.GetInstance(app, org, instanceOwnerId, instanceGuid);
+            if (instance == null)
+            {
+                return NotFound();
+            }
+
+            if (instance.Process != null)
+            {
+                return Conflict($"Process is already started. Use next");
+            }
+
+            ProcessKeeper = BpmnReader.Create(processService.GetProcessDefinition(org, app));
+
+            string validStartElement = GetValidStartEvenOrError(startEvent, out ActionResult startEventError);
+            if (startEventError != null)
+            {
+                return startEventError;
+            }
+
+            // trigger start event
+            Instance updatedInstance = await StartProcessOfInstance(org, app, instanceOwnerId, instanceGuid, instance, validStartElement);
+
+            // trigger next task
+            string nextValidElement = GetNextValidElementOrError(validStartElement, out ActionResult nextElementError);
+            if (nextElementError != null)
+            {
+                return nextElementError;
+            }
+
+            updatedInstance = await UpdateProcessStateToNextElement(org, app, updatedInstance, instanceOwnerId, instanceGuid, nextValidElement);
+
+            if (updatedInstance != null)
+            {
+                return Ok(updatedInstance.Process);
+            }
+
+            return StatusCode(500, $"Unknown error. Cannot change process state!");
+        }        
+
+        private string GetNextValidElementOrError(string currentElement, out ActionResult nextElementError)
+        {
+            nextElementError = null;
+            string nextElementId = null;
+
+            List<string> nextElements = ProcessKeeper.NextElements(currentElement);            
+
+            if (nextElements.Count > 1)
+            {
+                nextElementError = Conflict($"There are more than one element reachable from element {currentElement}");
+            }
+            else
+            {
+                nextElementId = nextElements.First();
+            }
+
+            return nextElementId;
+        }
+
+        private string GetValidStartEvenOrError(string proposedStartEvent, out ActionResult startEventError)
+        {
+            startEventError = null;
+            string validStartEvent = null;
+
+            List<string> possibleStartEvents = ProcessKeeper.StartEvents();
+
+            if (!string.IsNullOrEmpty(proposedStartEvent) && possibleStartEvents.Contains(proposedStartEvent))
+            {
+                validStartEvent = proposedStartEvent;
+            }
+            else if (possibleStartEvents.Count == 1)
+            {
+                validStartEvent = possibleStartEvents.First();
```

Comment:
should the user be informed that the start event they sent to the controller is not the one being used?

Reply:
OK, has reworked code to return error if start event is invalid.